pkg/apis/core/types.go

```diff
@@ -2511,6 +2511,11 @@ type PodSecurityContext struct {
 	// +k8s:conversion-gen=false
 	// +optional
 	HostIPC bool
+	// Use the host's user namespace.
+	// Optional: Defaults to empty.
```

Comment:
If the feature gate is not enabled, it should be left unset, so `empty` means status quo.  I think this text should update to align with `ShareProcessNamespace` which is clearer about how it handles defaults.  If the feature gate is disabled, this field is never set.  If the feature gate is enabled, the field defaults to `false`.  

It is worth discussing why the field defaults to `false` when enabled.  If we contrast this with `ShareProcessNamespace`, we did not enable the shared pid namespace among containers by default in a pod because not sharing the pid namespace was a) backwards compatibility concern, b) could break many existing containers that assumed pid 1, c) in many cases, isolating pid namespace was a desirable security trait.

For user namespace support, it is potentially a backwards compatibility concern to default to `false`, but I think the security benefits of defaulting to false outweigh the backward compatibility concern.  In addition, we have mechanisms to default to `true` if there is a signal that the application required it (used other host namespaces, non-namespaced capabilities, contained a privileged container, used a host path volume).  In the case where the above defaulting is truly insufficient, the user can set the flag explicitly.

Finally, opt-in to this feature requires a manual container runtime change across the cluster to enable userns remapping so it is very much a conscious decision.